src/main/java/uk/gov/hmcts/reform/managecase/repository/DefaultDefinitionStoreRepository.java

```diff
@@ -21,13 +20,21 @@ public DefaultDefinitionStoreRepository(DefinitionStoreApiClient definitionStore
     }
 
     @Override
+    @Cacheable(value = "challengeQuestions",
+        key = "{#caseTypeId, #challengeQuestionId, #root.target.getLatestVersion(#caseTypeId)}")
     public ChallengeQuestionsResult challengeQuestions(String caseTypeId, String challengeQuestionId) {
         return definitionStoreApiClient.challengeQuestions(caseTypeId, challengeQuestionId);
     }
 
     @Override
-    @Cacheable(value = "caseRoles", key = "#caseTypeId")
+    @Cacheable(value = "caseRoles",
+        key = "{#caseTypeId, #userId, #jurisdiction, #root.target.getLatestVersion(#caseTypeId)}")
```

Comment:
Can you remove userId and Jurisdiction form key please (at least userId). Otherwise this cache wouldn't be useful much.